src/component/handlers/composite-android/DraftEditorCompositionHandlerAndroid.js

```diff
@@ -62,6 +62,14 @@ const startCompositionTimeout = (editor) => {
     // If we are at a safe point to exit composition mode, do so to let renders through.
     const editorState = getEditorState(editor);
     if (editorState.isInCompositionMode()) {
+      if (hasMutation) {
+        setImmediate(() => {
```

Comment:
< ⭐️ I think we should use `setTimeout` 0 instead, `setImmediate` is nonstandard.

Reply:
Also, should we clear this timeout too in `cancelCompositionTimeout`?

Reply:
Do you know what this is waiting for? Why does it have to be done asynchronously?

Reply:
I tried both setTimeout and setImmediate and they both worked fine.  Draft actually uses `setImmediate` in a few places.  I can swap in `setTimeout` though, it's probably less surprising (and on android, I suspect they're equivalent).

Reply:
I don't think we need to clear the inner timeout, for it to happen you would need to exit composition mode at the exact instant after this code executed, and the browser would need to interleave the callback before the setTimeout.

That said, the impact would be that restoreDom is called twice, and I seem to remember it just swapping the key on the DraftEditor, which would likely end up at most causing a rerender.

Reply:
The timeout is tricky.  Without it, the browser will reset the cursor position if there was a mutation since the composition range updates, but doing it in a timeout means that Draft picks up on the next tick and resets the cursor to the correct position.